python/dgl/graphbolt/dataloader.py

```diff
@@ -226,6 +258,30 @@ def __init__(
                     Awaiter(Bufferer(feature_fetcher, buffer_size=1)),
                 )
 
+        if (
+            overlap_graph_fetch
+            and num_workers == 0
+            and torch.cuda.is_available()
+        ):
+            torch.ops.graphbolt.set_max_uva_threads(max_uva_threads)
+            samplers = dp_utils.find_dps(
+                datapipe_graph,
+                SamplePerLayer,
+            )
+            executor = ThreadPoolExecutor(max_workers=1)
+            for sampler in samplers:
+                datapipe_graph = dp_utils.replace_dp(
```

Comment:
It is a little bit weird to see FetcherAndSampler in the dataloader.py, but not sure what's the best way to do. I am worried about we can't maintain the complexity with all this dp replacement.

Is it possible to code this in neighbor_sampler.py in some way?

Reply:
It will lead to circular dependency in that case. I can remove FetcherAndSampler if you want and manually chain the datapipes here.

Reply:
Pushed my changes.

Reply:
Or, we can move the Waiter and Bufferer into a different file so that dataloader and the neighborsampler can import it.

Reply:
Moved the datapipe components into base.py and the FetcherAndSampler into sample_neighbors.py.